feat/fix-storeMenu,order

jvn1o opened the pull request:
## 📌 개요
- storeMenu,order 수정

## 🔨 변경 사항
- UserRole이 Manager, Master인 경우 softDelete 처리 된 메뉴들도 조회할 수 있게 로직 추가, 기존처럼 Owner는 본인의 StoreMenu는 softDelete 되지 않은 메뉴들만 조회할 수 있다.

- phoneNumber의 데이터 정책에 따른 수정

- address의 최소 10자 ~ 최대 255자 제약 조건에 따른 Test 수정

## ✅ 테스트
- [x] 단위/통합 테스트 통과 확인
- [x] 로컬 실행 후 정상 동작 확인

## ☑️ 체크리스트
- [x] 코드 컨벤션 준수
- [x] 문서화 (ERD, API 명세서 등) 반영

## 🙋 리뷰 요청사항(선택)
- (예시)이런 A라는 문제가 있는데, B와 C중에 어떤게 괜찮을까요?

jvn1o committed:
feat/fix-storeMenu-API

UserRole이 Manager, Master인 경우 softDelete 처리 된 메뉴들도 조회할 수 있게 로직 추가, 기존처럼 Owner는 본인의 StoreMenu는 softDelete 되지 않은 메뉴들만 조회할 수 있다.

jvn1o committed:
final Complete

- phoneNumber의 데이터 정책에 따른 수정

- address의 최소 10자 ~ 최대 255자 제약 조건에 따른 Test 수정

skfkgla commented on src/main/java/com/sparta/delivery/backend/store/menu/service/StoreMenuService.java:
```diff
@@ -90,8 +91,14 @@ public Page<ResGetListStoreMenuDto> getStoreMenusByStoreId(
+		Page<StoreMenu> storeMenuList;
+
+		// UserRole이 Manager || Master이면 softDelete 된 목록들도 조회할 수 있게 처리
+		if (user.getRole() == UserRoleEnum.MANAGER || user.getRole() == UserRoleEnum.MASTER) {
```
이것도 선택 할 수 있도록 하면 좋을 것 같습니다! 기본값은 삭제된 값은 안보이는 것으로 부탁드립니다 :)

jvn1o replied:
아 확인했습니다 수정하겠습니다~

jvn1o committed:
feat/fix-storeMenu-API

Manager, Master 모두 삭제된 메뉴들 안 보이도록 정책 결정

jvn1o committed:
feat/fix-order-api

- 고객이 주문 생성 후 5분 이내 && 요청 상태가 주문 중 일 때에만 취소 가능하도록 설계
- 고객이 주문 수락 행위를 하려고 하면 실패하는 테스트 케이스 추가
- OrderController에 Swagger 누락된 부분 추가
